### ISSUE TITLE
Function documentation ignored when placed under conditional

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#14962]
If you save the following code to a file and `include()` it, neither `?f` nor `?g` will print any documentation. Removing the `if` fixes the problem. This is annoying when only adding a function depending on the Julia version, notably when extending a Base function which may not exist in older releases.

```julia
if false
    """
    Do something funny.
    """
    f(x) = 1
else
    """
    Do something sad.
    """
    g(x) = 1
end
```

**Comment by yuyichao** [comment]
I thought this was intentional. The `@doc "..." ->` should still work?

**Comment by nalimilan** [comment]
Indeed it does. Maybe that's intentional and there are implications I'm not seeing, but that's certainly not user-friendly.

**Comment by tlnagy** [comment]
What was the original intent here?

**Comment by kshyatt** [comment]
@MichaelHatherly is this still a problem?

**Comment by MichaelHatherly** [comment]
Presumably covered by https://github.com/JuliaLang/julia/pull/20198?

**Comment by omus** [comment]
Still currently an issue. Can be seen in the wild with the [`LibGit2.Consts.GIT_CONFIG`](https://github.com/JuliaLang/julia/blob/e545c657d5ad88bc77ade254e96499977905e51c/base/libgit2/consts.jl#L349-L393) docstring.

**Comment by omus** [comment]
Documentation also [states that docstrings in `if`s should work](https://docs.julialang.org/en/latest/manual/documentation/#Advanced-Usage-1):
```julia
if condition()
    "..."
    f(x) = x
end
```

**Comment by bramtayl** [comment]
Yeah I wrote those docs and I was wrong, both about if and let blocks... Sorry...

**Comment by omus** [comment]
Not a big deal. I'm not sure if this is going to be changed but we should update the documentation. The working syntax is:
```julia
if true
    @doc """
    ...
    """ ->
    f(x) = x
end
```

**Comment by LilithHafner** [comment]
I support inserting @doc implicitly in these cases. An example in the wild is a `reinterpret` docstring https://github.com/JuliaLang/julia/blob/4ed419555a735534fae4343ec9f1114702b1ab12/base/reinterpretarray.jl#L27-L50

**Comment by LilithHafner** [comment]
```julia
if true
    begin
        """
        ...
        """
        f(x) = x
    end
end
```
works :P
